File: Java/Platform/Basics/ThreadTest/Program1.java
class Program {
	
	private static String client;

	private static void handleJob(int id) {
		System.out.printf("Thread<%x> has accepted job<%d> for %s%n", Thread.currentThread().hashCode(), id, client);
		Worker.doWork(10 * id);
		System.out.printf("Thread<%x> has finished job<%d> for %s%n", Thread.currentThread().hashCode(), id, client);
	}

	public static void main(String[] args) {
		int n = args.length > 0 ? Integer.parseInt(args[0]) : 1;
		client = "Jack";
		handleJob(n);
		client = "Jill";
		handleJob(2);
	}
}

File: dotnet/Platform/Basics/ThreadTest/DemoApp/Program1.cs
class Program
{
    private static string client;

    private static void HandleJob(int id)
    {
        Console.WriteLine("Thread<{0}> has accepted job<{1}> for client<{2}>", Thread.CurrentThread.ManagedThreadId, id, client);
        Worker.DoWork(10 * id);
        Console.WriteLine("Thread<{0}> has finished job<{1}> for client<{2}>", Thread.CurrentThread.ManagedThreadId, id, client);
    }   

    static void Main(string[] args)
    {
        int n = args.Length > 0 ? int.Parse(args[0]) : 1;
        client = "Jack";
        HandleJob(n);
        client = "Jill";
        HandleJob(2);
    }
}
